fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -0,0 +1,200 @@
+            String partitionFormat = DynamicPartitionUtil.getPartitionFormat(partitionColumn);
+
+            Calendar calendar = Calendar.getInstance();
+            TableProperty tableProperty = table.getTableProperty();
+            int end = Integer.parseInt(tableProperty.getDynamicPartitionProperty().getEnd());
+            for (int i = 0; i <= end; i++) {
+                String dynamicPartitionPrefix = tableProperty.getDynamicPartitionProperty().getPrefix();
+                String partitionRange = DynamicPartitionUtil.getPartitionRange(tableProperty.getDynamicPartitionProperty().getTimeUnit(),
+                        i, (Calendar) calendar.clone(), partitionFormat);
+                String partitionName = dynamicPartitionPrefix + DynamicPartitionUtil.getFormattedPartitionName(partitionRange);
+                // continue if partition already exists
+                if (table.getPartition(partitionName) != null) {
+                    continue;
+                }
+
+                // construct partition desc
+                String nextBorder = DynamicPartitionUtil.getPartitionRange(tableProperty.getDynamicPartitionProperty().getTimeUnit(),
+                        i + 1, (Calendar) calendar.clone(), partitionFormat);
+                PartitionValue partitionValue = new PartitionValue(nextBorder);
+                PartitionKeyDesc partitionKeyDesc = new PartitionKeyDesc(Collections.singletonList(partitionValue));
+                HashMap<String, String> partitionProperties = new HashMap<>(1);
+                partitionProperties.put("replication_num", String.valueOf(DynamicPartitionUtil.estimateReplicateNum(table)));
+                SingleRangePartitionDesc rangePartitionDesc = new SingleRangePartitionDesc(true, partitionName,
+                        partitionKeyDesc, partitionProperties);
+
+                // construct distribution desc
+                HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) table.getDefaultDistributionInfo();
+                List<String> distColumnNames = new ArrayList<>();
+                for (Column distributionColumn : hashDistributionInfo.getDistributionColumns()) {
+                    distColumnNames.add(distributionColumn.getName());
+                }
+                DistributionDesc distributionDesc = new HashDistributionDesc(
+                        Integer.parseInt(tableProperty.getDynamicPartitionProperty().getBuckets()), distColumnNames);
+
+                // add partition according to partition desc and distribution desc
+                AddPartitionClause addPartitionClause = new AddPartitionClause(rangePartitionDesc, distributionDesc, null);
+                try {
+                    Catalog.getInstance().addPartition(db, table.getName(), addPartitionClause);
+                    setDynamicPartitionState(State.NORMAL);
+                    setMsg(defaultValue);
+                } catch (DdlException e) {
```

Comment:
Better to add a `LOG.info` here, for tracing problem easily.

Reply:
Done